ledger/participant-state/src/main/scala/com/daml/ledger/participant/state/v1/SubmissionResult.scala

```diff
@@ -14,18 +14,11 @@ object SubmissionResult {
     override val description: String = "The request has been received"
   }
 
-  /** The system is overloaded, clients should back off exponentially */
-  case object Overloaded extends SubmissionResult {
-    override val description: String = "System is overloaded, please try again later"
-  }
-
-  /** Submission is not supported */
-  case object NotSupported extends SubmissionResult {
-    override val description: String = "Submission is not supported"
-  }
-
-  /** Submission ended up with internal error */
-  final case class InternalError(reason: String) extends SubmissionResult {
-    override val description: String = s"Failed with an internal error, reason=$reason"
+  /** The submission has failed with a synchronous error.
```

Comment:
```suggestion
  /** The submission has failed with a synchronous error.
  * 
  * Please see the documentation on the `CommandSubmissionService` to choose appropriate errors.
```

Reply:
We can add that now, but I do not expect that the documentation on the `CommandSubmissionService` will ever be exhaustive, as different ledger implementations will generate different kinds errors. So your suggestion is too strong. How about the following:
```suggestion
  /** The submission has failed with a synchronous error.
    *
    * See the documentation on the `CommandSubmissionService` for how to report common errors.
```